cpp/src/arrow/c/bridge.cc

```diff
@@ -566,15 +575,31 @@ struct ArrayExporter {
       --n_buffers;
       ++buffers_begin;
     }
+
+    bool need_variadic_buffer_sizes =
+        data->type->id() == Type::BINARY_VIEW || data->type->id() == Type::STRING_VIEW;
+    if (need_variadic_buffer_sizes) {
+      ++n_buffers;
+    }
+
     export_.buffers_.resize(n_buffers);
     std::transform(buffers_begin, data->buffers.end(), export_.buffers_.begin(),
                    [](const std::shared_ptr<Buffer>& buffer) -> const void* {
                      return buffer ? buffer->data() : nullptr;
                    });
 
+    if (need_variadic_buffer_sizes) {
+      auto variadic_buffers = util::span(data->buffers).subspan(2);
+      export_.variadic_buffer_sizes_.resize(variadic_buffers.size());
+      for (auto [size, buf] : Zip(export_.variadic_buffer_sizes_, variadic_buffers)) {
+        size = buf->size();
+      }
```

Comment:
the `size` here is a reference? is there any way for it to be more explicit that this is a reference in the syntax here?

Reply:
Yes, or perhaps simply avoid `Zip` here and use the more usual and more readable push-back approach:
```c++
      export_.variadic_buffer_sizes_.reserve(variadic_buffers.size());
      for (const auto& buf : variadic_buffers) {
        export_.variadic_buffer_sizes_.push_back(buf->size());
      }